app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/DownloadAttachmentViewModel.kt

```diff
@@ -43,26 +43,35 @@ class DownloadAttachmentViewModel @Inject constructor(
 
     private val ioCoroutineContext = viewModelScope.coroutineContext(ioDispatcher)
 
-    private val attachmentResource
-        inline get() = savedStateHandle.get<String>(DownloadAttachmentProgressDialogArgs::attachmentResource.name)!!
+    private val attachmentLocalUuid
+        inline get() = savedStateHandle.get<String>(DownloadAttachmentProgressDialogArgs::attachmentLocalUuid.name)!!
 
     /**
      * We keep the Attachment, in case the ViewModel is destroyed before it finishes downloading
      */
     private var attachment: Attachment? = null
 
     fun downloadAttachment() = liveData(ioCoroutineContext) {
-        val localAttachment = attachmentController.getAttachment(attachmentResource).also { attachment = it }
-        val attachmentFile = localAttachment.getCacheFile(appContext)
-        val isAttachmentCached = localAttachment.hasUsableCache(appContext, attachmentFile) ||
-                LocalStorageUtils.downloadThenSaveAttachmentToCacheDir(attachmentResource, attachmentFile)
+        val attachment = runCatching {
 
-        if (isAttachmentCached) {
-            emit(localAttachment)
-            attachment = null
-        } else {
-            emit(null)
-        }
+            val localAttachment = attachmentController.getAttachment(attachmentLocalUuid).also { attachment = it }
+
+            var isAttachmentCached = localAttachment.hasUsableCache(appContext, localAttachment.getUploadLocalFile())
```

Comment:
I can't because sometime it's a cache file that is passed, and it make no sense to pass a uploadFile (for example when clicking on an attachment from the threadFragment